Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PaalmanPingsMonteCarloAbsorption.py

```diff
@@ -0,0 +1,915 @@
+                                 'Chemical Formula.')
+        self.declareProperty(name='ContainerAttenuationXSection', defaultValue=0.0,
+                             validator=FloatBoundedValidator(0.0),
+                             doc='The absorption cross-section for the can material in barns. To be used instead of '
+                                 'Chemical Formula.')
+        self.declareProperty(name='ContainerDensityType', defaultValue='Mass Density',
+                             validator=StringListValidator(['Mass Density', 'Number Density']),
+                             doc='Use of Mass density or Number density for the container.')
+        self.declareProperty(name='ContainerNumberDensityUnit', defaultValue='Atoms',
+                             validator=StringListValidator(['Atoms', 'Formula Units']),
+                             doc='Choose which units ContainerDensity refers to. Allowed values: [Atoms, Formula Units]')
+        self.declareProperty(name='ContainerDensity', defaultValue=0.0,
+                             validator=FloatBoundedValidator(0.0),
+                             doc='The value for the container Mass density (g/cm^3) or Number density (1/Angstrom^3).')
+
+        self.setPropertyGroup('ContainerMaterialAlreadyDefined', 'Container Material')
+        self.setPropertyGroup('ContainerChemicalFormula', 'Container Material')
+        self.setPropertyGroup('ContainerCoherentXSection', 'Container Material')
+        self.setPropertyGroup('ContainerIncoherentXSection', 'Container Material')
+        self.setPropertyGroup('ContainerAttenuationXSection', 'Container Material')
+        self.setPropertyGroup('ContainerDensityType', 'Container Material')
+        self.setPropertyGroup('ContainerNumberDensityUnit', 'Container Material')
+        self.setPropertyGroup('ContainerDensity', 'Container Material')
+
+        self.setPropertySettings('ContainerMaterialAlreadyDefined', container_condition)
+        self.setPropertySettings('ContainerChemicalFormula', container_condition)
+        self.setPropertySettings('ContainerDensityType', container_condition)
+        self.setPropertySettings('ContainerDensity', container_condition)
+
+        # Output Workspace Group
+        self.declareProperty(WorkspaceGroupProperty(name='CorrectionsWorkspace',
+                                                    defaultValue='corrections',
+                                                    direction=Direction.Output,
+                                                    optional=PropertyMode.Optional),
+                             doc='Name of the workspace group to save correction factors')
+
+    def PyExec(self):
+
+        sample_wave_ws = self._convert_to_wavelength(self._sample_ws)
+        self._set_beam(sample_wave_ws)
+
+        if not self._container_ws:
+            sample_wave_ws.setSample(Sample())
```

Comment:
I think the best would be to assume by default that both shape and material are specified.
And only if one of them is not, then the user input properties should become mandatory.

Reply:
Do you mean that the algorithm should assume by default that the shape and material are already specified in the sample\container workspaces that the user supplies? This would be fine with me and it would avoid adding an extra parameter to act as the switch I mentioned in my earlier comment